src/analyzer/analyzersilence.h

```diff
@@ -13,15 +14,29 @@ class AnalyzerSilence : public Analyzer {
     bool initialize(TrackPointer pTrack,
             mixxx::audio::SampleRate sampleRate,
             int totalSamples) override;
-    bool processSamples(const CSAMPLE* pIn, const int iLen) override;
+    bool processSamples(const CSAMPLE* pIn, SINT iLen) override;
     void storeResults(TrackPointer pTrack) override;
     void cleanup() override;
 
+    /// returns the index of the first sample in the buffer that is above -60 dB
+    /// or -1 if no sample is found
+    static SINT findFirstSound(const CSAMPLE* pIn, SINT iLen);
+
+    /// returns the index of the last sample in the buffer that is above -60 dB
+    /// or -1 if no sample is found. signalStart can be set to a known sample above
+    /// -60 dB or -1 to start with the following index.
+    static SINT findLastSound(const CSAMPLE* pIn, SINT iLen, SINT signalStart);
+
+    /// Returns true if the first sound if found at the given frame and logs a warning message if not.
```

Comment:
This function does not belong to the analyzer. It must not care about beat grids and such.

Reply:
This is not about beat grid, it is just a verify function for the data it has formally analyzed. That's it. 

But anyway, if you have an idea for improvement, please propose.

Reply:
The log message mentions beat grids. This clearly shows why this does not belong to the analyzer. Interpreting the results in a higher-level context does not belong in here.

The whole function could be implemented in terms of the public API.

Reply:
I can rephrase the warning message to mention only the first sound. But that is not relay helpful, because the message itself is correct. 
I can imagine to send an signal form the caching reader worker thread that to the GUI thread. Than we can receive  the signal for instance in the player class and write the log there. But this feels over-engineered just because the message wording. I would like to postpone the GUI interaction to a future PR, and continue in small steps.